rfd/0061-tsh-aliases.md

```diff
@@ -69,20 +69,25 @@ in `$TELEPORT_HOME/config/config.yaml` using the following syntax:
 
 ```yaml
 aliases:
-    "<alias>": "<command>"
+  "<alias>": "<command>"
 ```
 
 The `<alias>` can only be a top-level subcommand. In other words, users can
 define a new `tsh mycommand` alias but cannot define `tsh my command` command.
 
 A few notes regarding the `<command>`:
 
-- Unlike `git` aliases explored above, proposal is to always require the binary
-  name instead of defaulting to `tsh` and requiring special syntax otherwise.
+- Similar to `git` aliases explored above, the commands are understood to be in
+  the context of the `tsh` binary. We will avoid the need for special
+  syntax (`!bash ... `) by adding a new `tsh` command, `tsh exec`, which will
+  be used by the aliases to execute external commands.
```

Comment:
@klizhentas @xinding33 Any thoughts on the `tsh exec` approach @Tener is proposing? The idea is basically to:

- Treat each alias as a sub-command of `tsh` instead of just feeding it directly to "exec" as was in my original proposal.
- Introduce new `tsh exec` subcommand that will execute the provided command (kinda like `docker/kubectl exec`) with some Teleport specific env vars set.

The benefit is that `tsh exec` may be useful on its own outside of aliases.

The downside (at least it is IMO) is that this is one more thing for users to learn. They'd need to use it to invoke execute external commands and access env. vars (e.g. `$TSH`) instead of regular "bash".

So you'd define an alias like

```
"connect": "exec -- -c '$TSH login $0 && $TSH ssh $1'"
```

instead of

```
"connect": "bash -c '$TSH login $1 && $TSH ssh $2'"
```